registration/app/registration/services/WindowsNotificationRegistrar.scala

```diff
@@ -13,28 +14,63 @@ import scalaz.std.option.optionSyntax._
 class WindowsNotificationRegistrar(hubClient: NotificationHubClient)(implicit ec: ExecutionContext)
   extends NotificationRegistrar {
 
+  val logger = Logger(classOf[WindowsNotificationRegistrar])
+
   override def register(lastKnownChannelUri: String, registration: Registration): Future[\/[Error, RegistrationResponse]] = {
-    hubClient.registrationsByChannelUri(registration.deviceId).flatMap {
-      case \/-(Nil) => createOrUpdateRegistration(lastKnownChannelUri, registration)
-      case \/-(_ :: _) => createOrUpdateRegistration(registration.deviceId, registration)
+    findRegistrations(lastKnownChannelUri, registration).flatMap {
+      case \/-(Nil) => createRegistration(registration)
+      case \/-(azureRegistration :: Nil) => updateRegistration(azureRegistration, registration)
+      case \/-(oneRegistration :: moreRegistration) => deleteAndCreate(oneRegistration :: moreRegistration, registration)
```

Comment:
actually List destructuring is not needed here since you're passing whole list do `deleteAndCreate`